cli/compose/loader/loader.go

```diff
@@ -427,9 +427,21 @@ func loadVolumes(source types.Dict) (map[string]types.VolumeConfig, error) {
 		return volumes, err
 	}
 	for name, volume := range volumes {
-		if volume.External.External && volume.External.Name == "" {
-			volume.External.Name = name
-			volumes[name] = volume
+		if volume.External.External {
+			template := "both external and %s are specified for volume %q. Only one is allowed."
```

Comment:
Thank you for the review. Yes, I think this reads a lot better. Will update the PR.